handler/src/test/java/io/netty/handler/ssl/ocsp/OcspTest.java

```diff
@@ -0,0 +1,551 @@
+                    .sslProvider(sslProvider)
+                    .build();
+            try {
+                ocspfySslContext(serverSslContext, serverOcspContext);
+
+                SslContext clientSslContext = SslContextBuilder.forClient()
+                        .sslProvider(sslProvider)
+                        .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                        .build();
+                try {
+                    ocspfySslContext(clientSslContext, clientOcspContext);
+
+                    EventLoopGroup group = new DefaultEventLoopGroup();
+                    try {
+                        LocalAddress address = new LocalAddress("handshake-" + Math.random());
+                        Channel server = newServer(group, address, serverSslContext, serverHandler);
+                        Channel client = newClient(group, address, clientSslContext, clientHandler);
+                        try {
+                            assertTrue("Something went wrong.", latch.await(10L, TimeUnit.SECONDS));
+                        } finally {
+                            client.close();
+                            server.close();
+                        }
+                    } finally {
+                        group.shutdownGracefully(1L, 1L, TimeUnit.SECONDS);
+                    }
+                } finally {
+                    ReferenceCountUtil.release(clientSslContext);
+                }
+            } finally {
+                ReferenceCountUtil.release(serverSslContext);
+            }
+        } finally {
+            ssc.delete();
+        }
+    }
+
+    private static Channel newServer(EventLoopGroup group, SocketAddress address,
+            SslContext context, ChannelHandler handler) {
+
+        ServerBootstrap bootstrap = new ServerBootstrap()
+                .channel(LocalServerChannel.class)
+                .group(group)
+                .childHandler(newHandler(context, handler));
+
+        return bootstrap.bind(address)
+                .syncUninterruptibly()
+                .channel();
+    }
+
+    private static Channel newClient(EventLoopGroup group, SocketAddress address,
+            SslContext context, ChannelHandler handler) {
+
+        Bootstrap bootstrap = new Bootstrap()
+                .channel(LocalChannel.class)
+                .group(group)
+                .handler(newHandler(context, handler));
+
+        return bootstrap.connect(address)
+                .syncUninterruptibly()
+                .channel();
+    }
+
+    private static ChannelHandler newHandler(final SslContext context, final ChannelHandler handler) {
+        return new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) throws Exception {
+                ChannelPipeline pipeline = ch.pipeline();
+
+                SSLEngine engine = context.newEngine(ch.alloc());
```

Comment:
consider using `context.newHandler(ch.alloc())` instead. This may produce some different behavior in the future (PR https://github.com/netty/netty/pull/6365) and I would like to make sure it still works at that time.